### ISSUE TITLE
DB issued 2 deltas in response to one add command

### ISSUE THREAD

**Issue by hacksoncode** [hallidev/delta-bot-four#23]
I sent an "add" command (verified only once) about this comment: https://www.reddit.com/r/changemyview/comments/spmntw/cmv_fake_woke_people_are_racists_themselves/hwg7q32/

I got two "processed successfully" DMs 30 seconds apart, and when I went to look, I found these two deltas on the same comment, which shouldn't be possible:

https://www.reddit.com/r/changemyview/comments/spmntw/cmv_fake_woke_people_are_racists_themselves/hwkp4c9/
https://www.reddit.com/r/changemyview/comments/spmntw/cmv_fake_woke_people_are_racists_themselves/hwkp6l5/

The section from the console output:

2022-02-11 23:36:10.2668|INFO|DB4QueueDispatcher|Queue remaining: (0 primary), (8 ninja)
2022-02-11 23:36:45.2027|INFO|PrivateMessageProcessor|Processing incoming private message from: hacksoncode, subject: add
2022-02-11 23:36:45.3936|INFO|PrivateMessageProcessor|PrivateMessageHandler created: DeltaBotFour.Infrastructure.Implementation.PrivateMessageHandlers.ModAddDeltaPMHandler
2022-02-11 23:36:47.7031|INFO|DeltaAwarder|---START AWARD DELTA--- -> user: iwfan53, comment: /r/changemyview/comments/spmntw/cmv_fake_woke_people_are_racists_themselves/hwg7q32/
2022-02-11 23:36:47.7031|INFO|DeltaAwarder|   ---Updating wiki (award)
2022-02-11 23:36:51.2229|INFO|DeltaAwarder|   ---Setting flair (award)
2022-02-11 23:36:51.4007|INFO|DeltaAwarder|   ---Updating deltaboards (award)
2022-02-11 23:36:55.5029|INFO|DeltaAwarder|   ---Adding delta to local db (award)
2022-02-11 23:36:55.5029|INFO|DeltaAwarder|   ---Updating DeltaLog (award)
2022-02-11 23:37:25.6677|INFO|DeltaAwarder|   ---Updating post sticky (award)
2022-02-11 23:37:25.8455|INFO|RedditState|Waiting for 8165ms before performing edit...
2022-02-11 23:37:34.2431|INFO|CommentReplier|DeltaBot edited a reply -> result: PostSticky link: /r/changemyview/comments/spmntw/cmv_fake_woke_people_are_racists_themselves/hwg8fmj/
2022-02-11 23:37:34.2431|INFO|DeltaAwarder|---END AWARD DELTA---
2022-02-11 23:37:35.2126|INFO|CommentReplier|DeltaBot replied -> result: SuccessDeltaAwarded link: /r/changemyview/comments/spmntw/cmv_fake_woke_people_are_racists_themselves/hwg7q32/
2022-02-11 23:37:38.2310|INFO|PrivateMessageProcessor|Done processing private message.
2022-02-11 23:37:38.2310|INFO|DB4QueueDispatcher|Queue remaining: (2 primary), (8 ninja)
2022-02-11 23:37:38.3474|INFO|CommentProcessor|Processing incoming comment: /r/changemyview/comments/sq505d/cmv_including_samegender_love_stories_is_no_more/hwkozf0/
2022-02-11 23:37:38.3474|INFO|CommentProcessor|Done processing comment.
2022-02-11 23:37:38.3474|INFO|DB4QueueDispatcher|Queue remaining: (1 primary), (9 ninja)
2022-02-11 23:37:38.4568|INFO|PrivateMessageProcessor|Processing incoming private message from: hacksoncode, subject: add
2022-02-11 23:37:38.6382|INFO|PrivateMessageProcessor|PrivateMessageHandler created: DeltaBotFour.Infrastructure.Implementation.PrivateMessageHandlers.ModAddDeltaPMHandler
2022-02-11 23:37:40.8979|INFO|DeltaAwarder|---START AWARD DELTA--- -> user: iwfan53, comment: /r/changemyview/comments/spmntw/cmv_fake_woke_people_are_racists_themselves/hwg7q32/
2022-02-11 23:37:40.8979|INFO|DeltaAwarder|   ---Updating wiki (award)
2022-02-11 23:37:44.3576|INFO|DeltaAwarder|   ---Setting flair (award)
2022-02-11 23:37:44.5201|INFO|DeltaAwarder|   ---Updating deltaboards (award)
2022-02-11 23:37:49.4971|INFO|DeltaAwarder|   ---Adding delta to local db (award)
2022-02-11 23:37:49.4971|INFO|DeltaAwarder|   ---Updating DeltaLog (award)
2022-02-11 23:37:52.4613|INFO|DeltaAwarder|   ---Updating post sticky (award)
2022-02-11 23:37:52.6449|INFO|RedditState|Waiting for 8086ms before performing edit...
2022-02-11 23:38:00.9722|INFO|CommentReplier|DeltaBot edited a reply -> result: PostSticky link: /r/changemyview/comments/spmntw/cmv_fake_woke_people_are_racists_themselves/hwg8fmj/
2022-02-11 23:38:00.9722|INFO|DeltaAwarder|---END AWARD DELTA---
2022-02-11 23:38:01.8479|INFO|CommentReplier|DeltaBot replied -> result: SuccessDeltaAwarded link: /r/changemyview/comments/spmntw/cmv_fake_woke_people_are_racists_themselves/hwg7q32/
2022-02-11 23:38:04.4477|INFO|PrivateMessageProcessor|Done processing private message.

**Comment by hallidev** [comment]
Yeah that's strange, will keep an eye in case in happens again. Just FYI - be careful when copy/pasting from the console. If you don't right-click when you're finished, the bot freezes. I just had to resume it because it went down.

**Comment by hacksoncode** [comment]
Hmmm... this isn't exactly the same because it didn't award multiple times... but I just sent an add PM and DB4 processed it 3 times, awarding the delta the first time and sending me an "I can't do that, Dave" message each of the other times. 

Add link: https://www.reddit.com/r/changemyview/comments/sxqt5y/cmv_i_probably_dont_really_know_what_the_word/hxu144y/

I wonder if there's some kind of systemic reddit problem with multiply sending PMs, or if it's a problem with the API, or API wrapper class, or...
